pkg/sql/distsql_running.go

```diff
@@ -100,6 +101,22 @@ type runnerResult struct {
 	err    error
 }
 
+type runnerDialErr struct {
```

Comment:
I like the approach of tagging errors that occur during the dial flow. This is a robust way to handle issues that occur during start up, which is the most common case we have seen for Serverless.